consensus/core/src/core.rs

```diff
@@ -847,10 +981,27 @@ mod test {
 
             last_round_blocks = this_round_blocks;
         }
+
+        for (core, _) in cores {
+            // Check commits have been persisted to store
+            let last_commit = core
+                .store
+                .read_last_commit()
+                .unwrap()
+                .expect("last commit should be set");
+            // There are 8 leader rounds with rounds completed up to and including
+            // round 10
+            // TODO: debug why only 7 leaders are committed even though all the
+            // blocks for round 10 are eventually accepted.
```

Comment:
Actually the blocks of round 10 are never accepted. The above loop is _producing_ blocks up to round 10, but the last round blocks do get feedbacked to Cores to get accepted. So basically all the Cores will include from round 10 on their own blocks, not the others, thus not commit the leader of round 8